app/repository/organisation.py

```diff
@@ -87,5 +87,10 @@ def get_organisation_config(db: Session, org: Organisation) -> OrganisationConfi
     )
 
 
-def get_all_organisations(db: Session) -> list[Organisation]:
-    return db.query(Organisation).all()
+def get_organisations(db: Session, allowed_corpora: list[str]) -> list[Organisation]:
+    return (
+        db.query(Organisation)
+        .join(Corpus, Corpus.organisation_id == Organisation.id)
+        .filter(Corpus.import_id.in_(allowed_corpora))
```

Comment:
Almost - whilst this handles the case of locking down the API based on the allowed corpora, we want to return all the results if the allowed_corpora list is == `[]`. This can be done super easily within this function luckily, simply by splitting the query up like so

```
query = (
        db.query(Organisation)
        .join(Corpus, Corpus.organisation_id == Organisation.id)
    )
if allowed_corpora != []:
    query = query.filter(Corpus.import_id.in_(allowed_corpora))
result = query.all()        
```

Reply:
(plus any corresponding test update)

Reply:
Ah, yes I thought you said something like this in the morning. Could you please explain what the rationale behind this is actually? Would it not be better to error in this case?

Reply:
It's to keep our API working. The longterm goal I believe was that we had an API that could be used by techies/researchers to access our data (as it's all open anyway). Our introduction of the app_token works for all requests generated from within a custom app, but don't work for standalone API calls when you don't know that you need to add in the app_token as a custom header - plus users can't currently generate their own custom app tokens.